app/Http/Controllers/Auth/LoginController.php

```diff
@@ -3,37 +3,130 @@
+     *  5) Secondary Authentication users are then logged in once completed successfully.
      */
-    protected $redirectTo = '/home';
 
-    /**
-     * Create a new controller instance.
-     *
-     * @return void
-     */
     public function __construct()
     {
         $this->middleware('guest')->except('logout');
     }
+
+    public function logout()
+    {
+        Auth::logout();
+        Auth::guard('partial_web')->logout();
+        return redirect('/');
+    }
+
+    /*
+     * Step 1: Redirect to VATSIM.NET SSO
+     */
+    public function loginWithVatsimSSO()
+    {
+        if(Auth::guard('partial_web')->check()){
+            return redirect()->route('login.secondary');
+        }
+
+        $sso = new VATSIMSSO();
+        return $sso->login(url('/login/sso/verify'), function ($key, $secret, $url) {
+            Session::put('vatsimauth', compact('key', 'secret'));
+            return redirect($url);
+        }, function ($error) {
+            throw new Exception('Could not authenticate: ' . $error['message']);
+        });
+    }
+
+    /*
+     * Step 2: Verify SSO Login after redirect
+     */
+    public function verifySSOLogin()
+    {
+        $sso = new VATSIMSSO();
+        $session = Session::get('vatsimauth');
+
+        return $sso->validate(
+            $session['key'],
+            $session['secret'],
+            Input::get('oauth_verifier'),
+            function ($vatsimUser) {
+
+                $user = User::firstOrNew(['id' => $vatsimUser->id]);
+                $user->name_first = utf8_decode($vatsimUser->name_first);
+                $user->name_last = utf8_decode($vatsimUser->name_last);
+                $user->email = $vatsimUser->email;
+                $user->experience = $vatsimUser->experience;
+                $user->joined_at = $vatsimUser->reg_date;
+                $user->last_login = Carbon::now();
+                $user->last_login_ip = Request::ip();
+                $user->inactive = $vatsimUser->rating->id == -1 ? true : false;
+                $user->save();
+
+                if ($user->hasPassword()) {
+                    Auth::guard('partial_web')->loginUsingId($vatsimUser->id);
+                    return redirect()->route('login.secondary');
+                }
+
+                Auth::loginUsingId($vatsimUser->id);
+                return redirect()->intended('/home');
+            },
+            function ($error) {
+                throw new Exception($error['message']);
+            }
+        );
+    }
+
+    /*
+     * Step (3): Show Secondary Sign In
+     */
+
+    public function showSecondarySignin()
+    {
+        $user = Auth::guard('partial_web')->user();
+
+        if (!$user->hasPassword()) {
+            return redirect('/');
+        }
+
+        return view('auth.secondary');
+    }
+
+    /*
+     * Step (4): Fully Authenticate
+     */
+
+    public function verifySecondarySignin(\Illuminate\Http\Request $request)
```

Comment:
Why do you need a different request class from the one already imported in the facade?

Reply:
Because the facade class =/= a Http request, but will update to make it neater